### PR TITLE
Don't abort listing tags when we encounter a digest

### PR DESCRIPTION
**Pull request by mtrmac**
Per https://github.com/containers/skopeo/issues/2346, that happens in the wild.

### CODE REVIEW

File: docker/docker_image.go [hunk 1]
```diff
@@ -90,6 +91,13 @@ func GetRepositoryTags(ctx context.Context, sys *types.SystemContext, ref types.
 		}
 		for _, tag := range tagsHolder.Tags {
 			if _, err := reference.WithTag(dr.ref, tag); err != nil { // Ensure the tag does not contain unexpected values
+				// Per https://github.com/containers/skopeo/issues/2346 , unknown versions of JFrog Artifactory,
+				// contrary to the tag format specified in
+				// https://github.com/opencontainers/distribution-spec/blob/8a871c8234977df058f1a14e299fe0a673853da2/spec.md?plain=1#L160 ,
+				// include digests in the list.
+				if _, err := digest.Parse(tag); err == nil {
+					logrus.Debugf("Ignoring invalid tag %q matching a digest format", tag)
```
Comment (rhatdan) [inline docker/docker_image.go:99]: This does not appear to be ignoring?

Comment (mtrmac) [reply docker/docker_image.go:99]: Doh!. Thanks.

**Comment by rhatdan** [comment]
LGTM
@giuseppe @vrothberg @Luap99 @mheon PTAL

**Review by giuseppe — APPROVED** [review]
/lgtm

**Comment by mtrmac** [comment]
Fix confirmed in https://github.com/containers/skopeo/issues/2346#issuecomment-2146389913 , for the record.

**Pull request merged**